### PR TITLE
fix: Explicitly drop metrics so runtime SIGINT guard drops last

### PR DESCRIPTION
**Pull request by stealthybox**
@ysndr suggested this in https://github.com/flox/flox/pull/1769#issuecomment-2260675622

SIGINT handler now works when metrics client is blocking on DNS.

issue: #1766

### CODE REVIEW

**Review by mkenigs — APPROVED** [review]

**Review by ysndr — CHANGES_REQUESTED** [review]
The sentry thing actually _has to_ be initialized before the runtime.
That was one the reason why we have a manual runtime creation in the first place

**Comment by mkenigs** [comment]
@ysndr can you summarize why so we can make that clear in comments?

**Comment by ysndr** [comment]
id recommend adding the explicit drop for our own metrics, or at least put this below the sentry init though that looks a bit weird so inbetween sentry and flox metrics :S

**Comment by ysndr** [comment]
@mkenigs @stealthybox from the sentry docs on this: https://docs.sentry.io/platforms/rust/#async-main-function

**Comment by stealthybox** [comment]
@ysndr nice catch -- updated comments and added explicit drops

**Review by ysndr — APPROVED** [review]
